feelslikemonday/app/src/main/java/com/example/feelslikemonday/DAO/UserDAO.java

```diff
@@ -0,0 +1,71 @@
+package com.example.feelslikemonday.DAO;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.example.feelslikemonday.model.User;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static androidx.constraintlayout.widget.Constraints.TAG;
+
+//To get the DAO, simply call UserDao.getInstance()
+//Uses the user email as the primary key
+//Uses documentation snippets from https://firebase.google.com/docs/firestore/manage-data/delete-data
+public class UserDAO {
+    private final FirebaseFirestore db = FirebaseFirestore.getInstance();
+
+    private static final UserDAO instance = new UserDAO();
+
+    private UserDAO(){}
+
+    public static UserDAO getInstance() {
+        return instance;
+    }
+
+    public void createOrUpdate(User user){
+        Map<String, Object> userMap = new HashMap<>();
+        userMap.put("object",user);
+
```

Comment:
wait so how do you ensure that you're over-riding a use and not duplicating if you're updating a user. Is  that what the line with the getEmail() is about below?

Reply:
iirc you cannot have duplicate ids in Firestore as far as I know